api/software.py

```
@@ -0,0 +1,346 @@
+    if not data:
+        return Response(status_code=204)
+
+    result = [{'name': row[0], 'count': int(row[1])} for row in data]
+    return CustomORJSONResponse({'success': True, 'data': result})
+
+
+@router.get('/v1/software')
+async def get_softwares(
+    category: str | None = None,
+    page: int = 1,
+    user: User = Depends(authenticate), # pylint: disable=unused-argument
```

Ruff (0.15.15)

[warning] Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)